node_modules/oae-principals/lib/test/util.js

```diff
@@ -1047,3 +1127,176 @@ var assertUpdateUserFails = module.exports.assertUpdateUserFails = function(rest
+ */
+var assertVerifyEmailSucceeds = module.exports.assertVerifyEmailSucceeds = function(restCtx, email, token, callback) {
+    // Determine whether the user is anonymous. The verifyEmail endpoint will sign the user in if
+    // they are anonymous and will return the user profile if they're not
+    RestAPI.User.getMe(restCtx, function(err, me) {
+        assert.ok(!err);
+        var wasAnonymous = me.anon;
+
+        RestAPI.User.verifyEmail(restCtx, email, token, function(err, user, response) {
+            assert.ok(!err);
+
+            // If the user was anonymous they should've been redirected to the /me page
+            if (wasAnonymous) {
+                assert.strictEqual(response.headers.location, '/me');
+
+            // Otherwise the user profile is returned
+            } else {
+                assert.strictEqual(user.emailVerified, true);
+                assert.strictEqual(user.email, email.toLowerCase());
+                assert.ok(!user.emailChange);
+            }
+
+            // Verify the email is now properly set and verified
+            RestAPI.User.getMe(restCtx, function(err, me) {
+                assert.ok(!err);
+                assert.ok(!me.anon);
+                assert.strictEqual(me.emailVerified, true);
+                assert.strictEqual(me.email, email.toLowerCase());
+                assert.ok(!me.emailChange);
+
+                // Assert the email address is also marked as verified when retrieving the user profile
+                RestAPI.User.getUser(restCtx, me.id, function(err, user) {
+                    assert.ok(!err);
+                    assert.strictEqual(user.emailVerified, true);
+                    assert.strictEqual(user.email, email.toLowerCase());
+                    assert.ok(!user.emailChange);
+
+                    return callback();
+                });
+            });
+        });
+    });
+};
+
+/**
+ * Assert that an email address can not be verified
+ *
+ * @param  {RestContext}    restCtx         The REST context of a user who can perform the request
+ * @param  {String}         email           The email address to verify
+ * @param  {String}         token           The token that can be used to verify the email address
+ * @param  {Function}       callback        Invoked when the verify operation fails as expected
+ * @param  {Object}         callback.err    An error that occurred, if any
+ * @throws {AssertionError}                 Thrown if the operation is unsuccessful or the expected results of verifying an email don't hold true
+ */
+var assertVerifyEmailFails = module.exports.assertVerifyEmailFails = function(restCtx, email, token, httpCode, callback) {
+    RestAPI.User.verifyEmail(restCtx, email, token, function(err, user, response) {
+        assert.ok(err);
+        assert.strictEqual(err.code, httpCode);
+        return callback();
+    });
```

Comment:
I think that it would still be a robust test to:

1. Get the Me profile, see if it's anonymous
2. If it was anonymous, ensure it is still anonymous afterward